### PR TITLE
views: make `If-Match` required on `put`

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

File: tests/test_views_item_put.py [hunk 1]
```diff
@@ -36,15 +36,7 @@ def test_valid_put(app, es, test_records, content_type, search_url,
...
-        assert res.status_code == 200
-
-        # Check that the returned record matches the given data
-        assert get_json(res)['metadata']['year'] == 1234
-        IndexFlusher(search_class).flush_and_wait()
-        res = client.get(search_url, query_string={"year": 1234})
-        assert_hits_len(res, 1)
-        # Retrieve record via get request
-        assert get_json(client.get(url))['metadata']['year'] == 1234
+        assert res.status_code == 400
```
Comment (michamos) [inline tests/test_views_item_put.py:39]: do you want to test the invalid or valid case here? name is `test_valid_put`, but you're testing the case when the header is not present. I think you probably want to test both.

Comment (drjova) [reply tests/test_views_item_put.py:39]: I'm testing both cases, 👍 the naming of the test was wrong.

Comment (michamos) [reply tests/test_views_item_put.py:39]: where's the valid put test now?

Comment (drjova) [reply tests/test_views_item_put.py:39]: https://github.com/inspirehep/invenio-records-rest/pull/10/files#diff-a8354eebb456ed950c17919d31071af0b95e5f6ca65e55f917635955d1ed0018L53

Comment (michamos) [reply tests/test_views_item_put.py:39]: link doesn't work for me

Comment (drjova) [reply tests/test_views_item_put.py:39]: oh it's a mess with github view, it's the next test if you expand the code: ``def test_valid_put_etag``

**Review by michamos — APPROVED** [review]

**Pull request merged**